# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: media/webrtc/signaling/src/peerconnection/PeerConnectionMedia.cpp
# Commit: 8196045d38b5
# Full Hash: 8196045d38b50e04313725215c36ba1618c50486
# Author: Byron Campen [:bwc] <docfaraday@gmail.com>
# Date: 2019-09-18 21:49:27
# Regressor Bug: 1569183
# File Overlap Count: 2
# Description:
#   Bug 1569183: Create an ipdl struct called WebrtcProxyConfig, to cut down on the arg count on lots of functions. r=mjf,mayhemer
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D45562
# ==============================================================================

diff -r 6e1eba5c88ad -r 8196045d38b5 media/webrtc/signaling/src/peerconnection/PeerConnectionMedia.cpp
--- a/media/webrtc/signaling/src/peerconnection/PeerConnectionMedia.cpp	Wed Sep 18 14:27:42 2019 +0000
+++ b/media/webrtc/signaling/src/peerconnection/PeerConnectionMedia.cpp	Tue Sep 17 18:19:04 2019 +0000
@@ -757,14 +757,9 @@
     const {
   MOZ_ASSERT(NS_IsMainThread());
 
-  NrSocketProxyConfig::ProxyPolicy proxyPolicy =
-      NrSocketProxyConfig::kEnableProxy;
-
-  if (mForceProxy) {
-    proxyPolicy = NrSocketProxyConfig::kForceProxy;
-  } else if (Preferences::GetBool("media.peerconnection.disable_http_proxy",
-                                  false)) {
-    proxyPolicy = NrSocketProxyConfig::kDisableProxy;
+  if (!mForceProxy &&
+      Preferences::GetBool("media.peerconnection.disable_http_proxy", false)) {
+    return nullptr;
   }
 
   nsCString alpn = NS_LITERAL_CSTRING("webrtc,c-webrtc");
@@ -773,6 +768,7 @@
     // Android doesn't have browser child apparently...
     return nullptr;
   }
+
   TabId id = browserChild->GetTabId();
   nsCOMPtr<nsILoadInfo> loadInfo = new net::LoadInfo(
       nsContentUtils::GetSystemPrincipal(), nullptr, nullptr, 0, 0);
@@ -780,8 +776,8 @@
   Maybe<net::LoadInfoArgs> loadInfoArgs;
   MOZ_ALWAYS_SUCCEEDS(
       mozilla::ipc::LoadInfoToLoadInfoArgs(loadInfo, &loadInfoArgs));
-  return std::unique_ptr<NrSocketProxyConfig>(
-      new NrSocketProxyConfig(id, alpn, *loadInfoArgs, proxyPolicy));
+  return std::unique_ptr<NrSocketProxyConfig>(new NrSocketProxyConfig(
+      net::WebrtcProxyConfig(id, alpn, *loadInfoArgs, mForceProxy)));
 }
 
 }  // namespace mozilla